src/prefect/infrastructure/docker.py

```diff
@@ -277,9 +279,31 @@ async def run(
         exit_code = container.attrs["State"].get("ExitCode")
         return DockerContainerResult(
             status_code=exit_code if exit_code is not None else -1,
-            identifier=container.id,
+            identifier=self._get_infrastructure_pid(container.id),
         )
 
+    async def kill(self, infrastructure_pid: str, grace_seconds: int):
+        docker_client = self._get_client()
+        base_url, container_id = self._parse_infrastructure_pid(infrastructure_pid)
+
+        if docker_client.api.base_url != base_url:
+            raise InfrastructureNotAvailable(
+                f"Unable to stop container {container_id!r}: the current docker api ",
+                f"base_url {docker_client.api.base_url!r} does not match the expected ",
+                f"api base_url {base_url}.",
```

Comment:
I try not to use variable names in messages e.g. "api base_url" should just be "API URL"